opcua/client/ua_client.py

```diff
@@ -173,7 +173,10 @@ def open_secure_channel(self, params):
 
         # FIXME: we have a race condition here
         # we can get a packet with the new token id before we reach to store it..
-        response = struct_from_binary(ua.OpenSecureChannelResponse, future.result(self.timeout))
+        try:
+            response = struct_from_binary(ua.OpenSecureChannelResponse, future.result(self.timeout))
+        except TimeoutError:
+            raise ConnectionError('OpenSecureChannelRequest() timeout')
```

Comment:
there is no try/except anywhere in that class, there is no reason to add one here. let users catch this (this is best practice)